internal/common/datarecord_test.go

```diff
@@ -359,3 +359,87 @@ func TestDataRecord_OriginName(t *testing.T) {
 		})
 	}
 }
+
+func TestDataRecord_RemainingBuffer(t *testing.T) {
```

Comment:
I think you should skip this test, you are adding 40 lines of boilerplate code for me to read for testing a getter of its simplest form.

Reply:
I'll leave it now to have time to merge this, and with the #48 , the getter would be removed anyhow.